src/components/Editor/index.js

```diff
@@ -220,6 +222,31 @@ class Editor extends Component {
     shortcuts.off(`CmdOrCtrl+${searchAgainKey}`);
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    if (Object.keys(nextProps).length != Object.keys(this.props).length) {
+      return true;
+    }
+    if (Object.keys(nextState).length != Object.keys(this.state).length) {
+      return true;
+    }
+
+    let key;
+
+    for (key in nextProps) {
+      if (key != "getExpression" && nextProps[key] != this.props[key]) {
+        return true;
+      }
+    }
+
+    for (key in nextState) {
+      if (nextProps[key] != this.props[key]) {
+        return true;
+      }
+    }
```

Comment:
Yeah, the more I think about the reselect work, I'm of a mind to do things _the right way_ unless this editor perf issue is showing up in regular usage.

For `Editor`, adding reselect could enable us to memoize the `getExpression` function prop and then this complex `shouldComponentUpdate` wouldn't be needed at all.